src/esrs/store.rs

```diff
@@ -9,20 +9,29 @@ use uuid::Uuid;
 
 use crate::esrs::SequenceNumber;
 
+/// An EventStore is responsible for persisting events that an aggregate emits into a database, and loading the events
+/// that represent an aggregate's history from the database.
 #[async_trait]
 pub trait EventStore<Event: Serialize + DeserializeOwned + Send + Sync, Error> {
+    /// Loads the events that an aggregate instance has emitted in the past.
     async fn by_aggregate_id(&self, id: Uuid) -> Result<Vec<StoreEvent<Event>>, Error>;
 
+    /// Persists multiple events into the database.  This should be done transactionally - either
+    /// all the events are persisted correctly, or none are.
+    ///
+    /// Persisting events may additionally trigger configured Policies or Projectors.
     async fn persist(
         &self,
         aggregate_id: Uuid,
-        event: Event,
-        sequence_number: SequenceNumber,
-    ) -> Result<StoreEvent<Event>, Error>;
+        events: Vec<Event>,
```

Comment:
I was thinking about using an `impl Iterator<Item = Event>` instead of a materialized Vec. But probably overkill, since the events should not be too many. And also because of the borrow checker nightmare that this could become :sweat_smile:

Reply:
yeah - we can 100% look at this in future if it's needed. I didn't want to add yet another generic to the store!

Reply:
Lol agree with you